### PR TITLE
improve static_show of unparseable Symbols

### PR DESCRIPTION
**Pull request by JeffBezanson**
Fixes symbol quoting logic in `static_show`.

### CODE REVIEW

**Review by vtjnash — APPROVED** [review]

**Comment by tkelman** [comment]
can this be tested?

**Comment by JeffBezanson** [comment]
Sure, but this is basically debugging code and we don't really use it for anything serious, so I don't believe it has any tests.

**Comment by JeffBezanson** [comment]
Tests added. Turns out we did have one static_show test, but it was in `spawn`, so I moved it to `show` which makes a bit more sense to me.

File: test/show.jl [hunk 1]
```diff
@@ -742,3 +742,25 @@ end
...
 @test sprint(Base.show_supertypes, Int64) == "Int64 <: Signed <: Integer <: Real <: Number <: Any"
 @test sprint(Base.show_supertypes, Vector{String}) == "Array{String,1} <: DenseArray{String,1} <: AbstractArray{String,1} <: Any"
+
+# static_show
+
+function static_shown(x)
+    p = Pipe()
+    Base.link_pipe(p; julia_only_read=true, julia_only_write=true)
+    ccall(:jl_static_show, Void, (Ptr{Void}, Any), p.in, x)
+    close(p.in)
```
Comment (vtjnash) [inline test/show.jl]: this is a deadlock condition. you can't write to a stream that doesn't have a reader

**Comment by vtjnash** [comment]
Somewhat non-obvious, but https://github.com/JuliaLang/julia/pull/16917/commits/ca6b83eb1941eec689e0a8766beeb84914b9ccd6#diff-21ef36be0213824dd286b390e3590742 also added some test coverage of this logic, since this fallback show logic is used to implement CoreIO.

File: test/show.jl (hunk 1 above)
Comment (vtjnash) [inline test/show.jl]: (fix is to make this `@async close(p.in)`)

**Comment by JeffBezanson** [comment]
Ok that's good; at least they're all in the same file now.

**Pull request merged**